src/System.IO.FileSystem/tests/Directory/GetFiles.cs

```diff
@@ -164,5 +166,24 @@ public override string[] GetEntries(string path, string searchPattern, SearchOpt
         {
             return Directory.GetFiles(path, searchPattern, option);
         }
+
+        [Theory, MemberData(nameof(TrailingSeparators))]
+        public void DirectoryWithTrailingSeparators(string trailing)
+        {
+            // When getting strings back we should retain the root path as specified for Directory.
+            // DirectoryInfo returns the normalized full path in all cases.
+
+            string root = GetTestFilePath() + (IsDirectoryInfo ? "" : trailing);
```

Comment:
Do we want to run this test at all if !IsDirectoryInfo? i.e. is it testing anything new in that case?  Seems like we're going to end up running this test N times now for each !IsDirectoryInfo case, all of which will be identical to each other, and none of which will be necessary?

Reply:
I'm testing both cases here. DirectoryInfo/Directory differ in what they return- Directory specifically is the one that regressed. In either case we don't have any tests describing the behavior of trailing separators in the specified path.

Reply:
Ok.  That makes sense for DirectoryInfo vs Directory.  Still is odd though that we end up executing this test N times for DirectoryInfo but in each such case the test does exactly the same thing, as the `trailing` value is ignored and replaced by `""`.

Reply:
They're flipped below. I'm checking that DirectoryInfo _doesn't_ return the wonky separators you pass in to enumeration. Creation doesn't care about it, and I'm taking advantage of that to unify the cases (and perhaps get a little extra coverage of that fact). I'll update the comments to explain it more as it isn't super obvious.

Reply:
Aha.  Ok, thanks.